### DISCUSSION TITLE
Object state in Zenoh

### DISCUSSION THREAD

**Discussion by smnrgrs** [eclipse-zenoh/roadmap#150] [Q&A] [answered]
Hi, is there a way to mark published Zenoh data as invalid/deleted/disposed so that subscribers know that it is no longer valid or no longer exists?

**Comment by gabrik** [comment, marked as answer]
Hi @smnrgrs, you can use the `delete` method: https://docs.rs/zenoh/latest/zenoh/prelude/sync/struct.Session.html#method.delete passing a key expression.
The subscribers will receive a `Sample` with `kind` delete: https://docs.rs/zenoh/latest/zenoh/prelude/enum.SampleKind.html

**Reply by smnrgrs** [reply]
Thanks!
